lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -20,15 +21,29 @@ class PokemonOverviewPage extends StatelessWidget {
         centerTitle: true,
         backgroundColor: primaryColor,
       ),
-      body: GridView.builder(
-        padding: const EdgeInsets.all(10.0),
-        itemCount: pokemons.length,
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
-        itemBuilder: (_, index) {
-          final pokemon = pokemons[index];
-          return PokemonCard(pokemon: pokemon);
+      body: pokemons.when(
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (errorMessage) {
+          _showErrorMessageSnackbar(context, message: errorMessage);
+          return const Center(child: Text(noPokemonsAvailableLabel));
         },
+        (data) => GridView.builder(
+          padding: const EdgeInsets.all(10.0),
+          itemCount: data.length,
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
+          itemBuilder: (_, index) {
+            final pokemon = data[index];
+            return PokemonCard(pokemon: pokemon);
+          },
+        ),
       ),
     );
   }
+
+  void _showErrorMessageSnackbar(BuildContext context, {required String? message}) {
+    final snackBar = SnackBar(content: Text(message.toString()));
```

Comment:
use null coalescing and use `emptyString` constant as its fallback value instead of using .toString() here to be safer 🙂